joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MiscTests.scala

```diff
@@ -269,4 +269,26 @@ class MiscTests extends RubyCode2CpgFixture {
       cpg.typeDecl.name("some_class").size shouldBe 1
     }
   }
+
+  "CPG for code with method ending with =" should {
+    val cpg = code("""
+        |class MyClass
+        | def foo=(value)
+        | puts value
+        | end
+        |end
+        |
+        |obj = MyClass.new
+        |obj.foo="arg"
+        |""".stripMargin)
+
+    "recognise all call nodes" in {
+      cpg.call.name("puts").size shouldBe 1
+      cpg.call.name("<operator>.fieldAccess").size shouldBe 1
+    }
+
+    "recognise all method nodes" in {
+      cpg.method.name("foo").size shouldBe 1
```

Comment:
Why not name it `foo=`? I think one use-case when using Joern would be to find assignment methods, and for that we could build a query to filter method names ending in `=`.

Reply:
@xavierpinho as far I think, `=` doesn't become part of the method name. When we call the method, we would just make use of `foo` and not `foo=`. I guess name should be `foo` only, having said that in order to handle the use case that you are pointing at, we need to introduce another property which will indicate if the method has been defined with `=` operator.

Reply:
@pandurangpatil 
It is a assignment method and so we can choose to represent it so.

@xavierpinho 
I have included = in the name. That avoids the crash and unblocks us for further testing. However, more changes will be needed to have the right call node to make the data flow work and I intend to do that in a separate PR

Reply:
@rahul-privado Thanks! Please add some comment or unit-test explaining that `x.foo=arg` should actually mean `x.foo(arg)` in these cases, and we can merge this.

Reply:
Added comments

Reply:
@rahul-privado It looks like Ruby considers them as setter methods. New learning for me, 

@xavierpinho sorry for the misunderstanding. I didn't go trough the entire context, I mistook it for the Ruby 3 syntax to define method with `=` operator.